2023/naughtylist.pod

```diff
@@ -0,0 +1,190 @@
+Title: Santa's Christmas Dancer2 Web App - Naughty & Nice List Manager
+Topic: HMLX and Dancer2
+Author: Ken Town <ktown007@gmail.com>
+
+=encoding utf8
+
+=pod 
+
+Ho, ho, ho! In this hilarious Christmas-themed Dancer2 web app tutorial, we'll help Santa manage his Naughty & Nice list. This time, we're using Perl's Dancer2 framework, but Santa's Christmas spirit is universal. Let's dive in and create Santa's very own web app!
+
+
+=head2 todo.pl
+
+    !/usr/bin/env perl
+
+    use Dancer2;
+    set template => 'mustache';
+
+        #Initialize Santa's Naughty & Nice List
+
+    my $list = { list => [] };
+
+    get '/' => sub {
+        send_file '/index.html';
+    };
+
+    any '/delete' => sub {
+        my $q = request->params;
+        my $index = 0;
+
+        # Find the naughty or nice kid to remove
+        $index++ until $list->{list}[$index]{task} eq $q->{task};
+        splice(@{$list->{list}}, $index, 1);
+        
+        # Display the updated list
+        template 'list' => $list;
+
+    };
+
+    any '/add' => sub {
+        my $q = request->params;
+
+        # Add a new kid to the list
+    push @{$list->{list}}, $q;
+    
+    # Display the updated list
+    template 'list' => $list;
+
+    };
+
+    any '/list' => sub {
+        # Display the current list
+        template 'list' => $list;
+    };
+
+    start;
+
+
+Here's what Santa's Christmas Dancer2 web app does:
+
+=over
+
+=item 1.
+
+We set up our Dancer2 app with Mustache templates.
+
+
+=item 2.
+
+Santa initializes his Naughty & Nice List as an empty array of tasks.
+
+
+=item 3.
+
+When you visit the root URL /, it serves an index.html file. Make sure to create this HTML file for a nice, festive interface.
+
+
+=item 4.
+
+To delete a naughty or nice kid, we look for the kid's task in the list and remove them. Then, we display the updated list using the 'list' template.
+
```

Comment:
Are both naughty and nice in the same list? I'd like to see how the app distinguishes between those two states. Maybe it's there and just not apparent to me.

Reply:
it is just a simple to do list with only the one field. The article added the Christmas theme and gave a purpose for the list :) so Timmy- nice and Jane - naughty etc works. I should do a screen shot like this.

Reply:
That's something you should explain in the text, but, it wouldn't be hard to have two lists. :)